robot/power.py

```diff
@@ -54,8 +54,8 @@ def start_button_pressed(self):
         return status["start-button"]
 
     def buzz(self, duration, note=None, frequency=None):
-        if note is None and frequency is None:
-            raise ValueError("Either note or frequency must be provided")
+        if bool(note) == bool(frequency):
+            raise ValueError("Please only provide note or frequency")
```

Comment:
This allows the caller to use this API as the following, which is likely confusing and silly. I'm also not convinced that this check is as clear as it could be (though it is clever).

``` python
board.buzz(4, False, 4)
```

Note that  this will also then fail since the check `note is not None` will be truthy and the user will get the possibly confusing `KeyError("False is an invalid note")`.

I suggest going back to the previous check (though maybe that could raise a `TypeError` instead?) and adding a separate check that at least one is provided.